[wrangler] Perform old-asset-ttl in parallel

shahyar opened the pull request:
* Rather than expiring each asset synchronously, use Promise.all

Fixes #4414

**What this PR solves / how to test:** Run multiple `deploy` with `--old-asset-ttl N` set.

**Author has addressed the following:**

- Tests
  - [ ] Included
  - [x] Not necessary because: Existing functionality covers this
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Associated docs
  - [x] Issue(s)/PR(s): #4414 #3384 #3385
  - [ ] Not necessary because:

penalosa reviewed (commented):
This looks good, thanks for adding it! Could you rebase your branch on main to pull in the latest tests?

shahyar commented:
Done.

petebacondarwin reviewed (changes requested):
I'm not sure that this is a good idea. If you had 1000s of assets to update then this would try to create 1000s of parallel requests, right?
Perhaps we could do this but be a bit more sophisticated by running batches in parallel?
(Or maybe I am missing something here...)

shahyar commented:
Indeed it would. What's a "sane" amount of requests? I can simply add `if (promises.length > LIMIT) { await promises; promises.length = 0; }` into the loop to batch it into sections.

petebacondarwin commented:
> Indeed it would. What's a "sane" amount of requests? I can simply add `if (promises.length > LIMIT) { await promises; promises.length = 0; }` into the loop to batch it into sections.

Yeah I think this would be a fair solution. I think LIMIT = 4 is probably about right. Generally most clients have a hard limit of 16 concurrent requests anyway, and at some point the benefit of parallelisation is lost since our backend API will become overloaded itself.

petebacondarwin commented:
@shahyar - can you update the PR accordingly? Or if you are not needing this change any more, we can just close the PR.

lrapoport-cf commented:
hey @shahyar  :) we haven't heard from you in a while so we’re going to close this pr for now, but please feel free to re-open if/when you're able to update per @petebacondarwin 's comment above and we will review. thanks!

shahyar commented:
Hey @lrapoport-cf @petebacondarwin -- apologies for the very, very late reply! I've rebased and added in a `PARALLEL_EXPIRE_LIMIT = 5`.

I'm unable to reopen the PR, but if you do, the latest code from my branch should get pulled in.

petebacondarwin commented:
Unfortunately, I can't reopen this either. It says this is because the `main` branch was force-pushed or recreated (I guess since this PR)? Please can you create a new PR with your rebased branch?

petebacondarwin commented:
Actually now that we have Workers + Assets working natively, can you consider using that rather than the this old legacy approach? We don't indend to maintain the legacy assets uplads going forward.
